x/mongo/driver/connstring/connstring.go

```diff
@@ -232,7 +233,7 @@ func (p *parser) parse(original string) error {
 		if strings.Contains(username, "/") {
 			return fmt.Errorf("unescaped slash in username")
 		}
-		p.Username, err = url.QueryUnescape(username)
+		p.Username, err = url2.Unescape(username, url2.EncodeUserPassword)
```

Comment:
Looking at the `"net/url"` code for [unescape](https://cs.opensource.google/go/go/+/refs/tags/go1.18.3:src/net/url/url.go;l=200;drc=55590f3a2b89f001bcadf0df6eb2dde62618302b), the behavior when mode is `encodeUserPassword` and when mode is `encodePathSegment` are identical. The already-exported [PathUnescape](https://pkg.go.dev/net/url#PathUnescape) calls `unescape` with mode `encodePathSegment`, which should behave identically to calling the now-exported `Unescape` with mode `EncodeUserPassword`.

Update this to call `url.PathUnescape` instead of `url2.Unescape`.